### ISSUE TITLE
Add support for Swift Package Manager

### ISSUE THREAD

**Issue by EmDee** [bugsnag/bugsnag-cocoa#407]
### Description
I think pretty self explanatory. Not sure if there is anything blocking this, if wanted I can take a look at this.

**Comment by kattrali** [comment]
Thanks for the report, @EmDee. I took a quick look at adding a package manifest on a branch in https://github.com/bugsnag/bugsnag-cocoa/commit/da7c0d6d7c053b7d152e744e8cf486a3470f0b44 - while I can get the library to build with `swift build`, I haven't managed to get Xcode to include it correctly when using `File > Swift Packages > Add Package Dependency` with:
```
URL: https://github.com/bugsnag/bugsnag-cocoa
branch: kattrali/spm
```

I think more needs to be done to expose the public headers in the `Source/` directory, but that's what I have so far if you want to take a look. I'm also a few Xcode11 betas behind, so it might be something fixed by updating.

**Comment by gabors** [comment]
Hi guys. Have you had any time to work on SwiftPM? Would love to use your service in a few high profile iOS, tvOS and macOS Catalyst projects in the digital media distribution field.

**Comment by jeffblake** [comment]
Bugsnag is the last dependency to go before I can drop Cocoapods :)

**Comment by gabors** [comment]
@kattrali. I did try the spm branch yesterday. The dependency builds fine but none of the Buhsnag classes are visible in swift code.

**Comment by kattrali** [comment]
I spent some time trying different combinations of arguments to `publicHeaderPaths` And tinkered with the module map but haven’t managed to get it working either. I’m stuck in the same spot - it builds but then no headers are exposed. I went digging through the SPM sources but don’t know what small thing is missing.

**Comment by EmDee** [comment]
Sorry, I haven't had much time to look into this. I have no experience with SPM whatsoever, but will see what I can do this weekend.

**Comment by gabors** [comment]
How is this coming along @EmDee @kattrali ?

**Comment by jeffblake** [comment]
Agreed, would love to see this. I have Bugsnag installed manually for the time being, currently banging my head against a wall with "No such module" error when archiving

**Comment by dtopolansky** [comment]
This would be great to see. Please can we get this functionality 🙏🏼I'm moving my entire project to use SPM and this is a bit of a blocker.

**Comment by jeffblake** [comment]
Love bump

**Comment by immortal79** [comment]
Any news ?

**Comment by JorisBodin** [comment]
Today new iOS projects are made in SPM. We pay for your product. We have to create a PR?!

**Comment by RyuX51** [comment]
I'd really love see SPM support as well, especially because you can't use CocoaPods and SPM together. I'd love to drop CocoaPods in favour of SPM.

**Comment by mattdyoung** [comment]
Hi!

Just an update from Bugsnag to say we're planning to work on this very soon! We're already working on a new major version of bugsnag-cocoa which is a few weeks away from release and will be aiming to incorporate support for Swift Package Manager into this as well.

**Comment by maxcan** [comment]
Investigating swift crash reporters.. not supporting swiftpm is a somewhat significant con for bugsnag :(

**Comment by michael-mckenna** [comment]
Looking forward to the updates. We are wanting to migrate to SwiftPM soon-ish

**Comment by garsdle** [comment]
@mattdyoung any news? I'm holding off moving to a different tool in hopes SPM will be supported soon. Thanks for the updates.

**Comment by johnkiely1** [comment]
Hi,
Unfortunately nothing new to report in terms of definite timelines. We have had a number of other items take priority so this has slipped a little. This is very much on our radar and we are planning to work on this shortly.

**Comment by mikemee** [comment]
@johnkiely1 thanks for the update.

FWIW, we stopped using BugSnag in large part because this is missing.

**Comment by livestrong** [comment]
I've gotten my local obj-c targets to work nicely with SPM. The headers like you mentioned were the tricky part.  This section helped me: https://github.com/apple/swift-package-manager/blob/master/Documentation/Usage.md#creating-c-language-targets.  I'm happy to take a look at your Package.swift file

**Comment by gsabran** [comment]
Any news? SPM has now been Apple's preferred way to package libraries for a long time now. Having a dependency not available on this is a blocker for any migration.

**Comment by jeffblake** [comment]
Thanks Joe for putting up a pull request! Hoping it gets a swift merge and publish. Almost a year to rename a few files....

**Comment by phillipsam** [comment]
Released in bugsnag-cocoa [6.1.0](https://github.com/bugsnag/bugsnag-cocoa/releases/tag/v6.1.0)
